### ISSUE TITLE
Bug: the report_id and the UUID embedded in any body_url blobstore filename, should always be the same.

### ISSUE THREAD

**Issue by jimduff-usds** [CDCgov/prime-reportstream#1418]
See also #1416 and #1010 - this is part of a series.  @MauriceReeves-usds , note that this is a bug in the new Filename template code.

The UUID embedded in the filename of a file in our blobstore should be the same as the UUID of the reportId.

You can see this problem easily by just doing this query:

```
psql prime_data_hub
select next_action, created_at, report_id, body_url from report_file order by created_at desc;
```

You'll see that the UUID in the blob_url is different from the report_id.  Here's a couple examples from my local:
```
 send        | 2021-06-28 08:11:00.33635-04  | 4252fdf0-575a-4c04-a8e1-9a007802df2d | http://azurite:10000/devstoreaccount1/reports/fl-covid-19-d238155a-c874-49a5-91dd-f0e770cd5d86-20210628121100.hl7
 batch       | 2021-06-28 08:10:34.460923-04 | 3cbc7559-3736-45e8-bc74-64cebe9c5494 | http://azurite:10000/devstoreaccount1/reports/fl-covid-19-5f68bec3-3727-4e42-803b-828e6d3cca70-20210628121034.internal
```

The report_id and the UUID embedded in any body_url blobstore filename, should always be the same, but they are not.  This issue was introduced in early June, and the problem is on `FilenameTemplate.kt` line 53, where a new UUID is being generated fresh, instead of using the UUID of the Report.kt object.

So, your task, should you choose to accept it, is to figure out how to pass the true/proper UUID down into that method.    I presume you'll do this by adding the UUId to the `translatorConfig` object, but that might be tricky - that obj seems to come from the yaml receiver settings?   So please study the code carefully and understand it before you start making changes.

NOTE:  this is confusing, but the above UUID rule does NOT apply to `external_name.`  For reports created by the 'send' step, with a non-null report_file.external_name, the external_name must be the same as the body_url filename of the the _parent_ report, created by the batch step.     This is important to ensure that filenames are completely identical for users who can get files two ways: 1) download site files (created by the batch step) or  2) sftp files (created by the send step).  It would be super confusing if the same data had two different filenames!

So after you fix the above issue, do some full runs, then look in the database and confirm that the external_name in the report_file entry from the `send` step is identical to the blobstore filename in the report_file entry from the `batch` step.

Better yet, you could add a query that confirms this trickiness to TestReportStream.kt, and add it as part of the assertions made in our `end2end` test.

**Comment by carlosfelix2** [comment]
I look in my local instance and in staging to see samples of blob files that do not have the correct report ID and I could not find any.  I used the following SQL statement to search:
`select next_action, created_at, position(report_id::text in body_url), report_id, body_url from report_file where body_url is not null AND position(report_id::text in body_url) is NULL order by created_at desc;`
This may have been fixed by the changes made in #1416 and #1010

**Comment by jimduff-usds** [comment]
Good catch.   This appears to be fixed, so I'm closing it.
